### ISSUE TITLE
Horizon Zero Dawn 1.11 update

### ISSUE THREAD

**Issue by phush0** [HansKristian-Work/vkd3d-proton#941]
After 1.11 update there introduced upscallers. With VKD3D from Proton Experimental - DLSS is not working. Nvidia Vulkan Beta 470.62.13. Startup params:

`PROTON_HIDE_NVIDIA_GPU=0 VKD3D_CONFIG=dxr,dxr11,force_static_cbv PROTON_ENABLE_NVAPI=1 gamemoderun mangohud prime-run %command%`

There are flickering textures in benchmark, on foliage (attached to buildings).

**Comment by jp7677** [comment]
Hi @phush0  could you try dxvk-nvapi from https://github.com/jp7677/dxvk-nvapi/pull/60 (there are binaries attached to that PR)?
May the game just stumbles over that one method when probing DLSS requirements.

**Comment by K0bin** [comment]
More problems:
https://github.com/ValveSoftware/Proton/issues/4125#issuecomment-989166246

**Comment by Vaudrain** [comment]
> Hi @phush0 could you try dxvk-nvapi from [jp7677/dxvk-nvapi#60](https://github.com/jp7677/dxvk-nvapi/pull/60) (there are binaries attached to that PR)? May be the game just stumbles over that one method when probing DLSS requirements.

Hi @jp7677 , I've attempted to do this (I believe I retrieved the correct binaries). The new nvapi doesn't cause DLSS to appear where it should in the menu, even with appropriate launch settings that function for other games. Looking at proton logs, it seems that the NvAPI workaround is still being enabled.

I don't believe the flickering textures/foliage are related to nvapi, as there are reports in [this steam thread](https://steamcommunity.com/app/1151640/discussions/0/3193611900735211185/) of the same issue presenting itself to users with AMD cards. The issue definitely does seem to be with proton, however.

Here are two proton logs. [In the first](https://github.com/HansKristian-Work/vkd3d-proton/files/7681319/steam-1151640-settings-DLSS-check.log), I booted the game, went to the settings menu, saw no DLSS, and closed.
[In the second](https://github.com/HansKristian-Work/vkd3d-proton/files/7681320/steam-1151640-gameplay.log), I loaded into an area dense with foliage, ran around a little, engaged in some combat and mount riding, saved, stood for a few moments, then exited the game.

**Comment by Blitzwing1** [comment]
Can definitely confirm that the graphical issues as described by @Vaudrain appear to be exactly the same on two separate systems. With the difference being that the Nvidia system doesn't open up DLSS when "PROTON_ENABLE_NVAPI=1" is enabled.

System 1.
OS: Pop!_OS 21.04
CPU: Intel 10750H
GPU Nvidia RTX 3070, 495.44
Proton Experimental

System 2.
OS: Pop!_OS 21.04
CPU: Ryzen 7 3700X
GPU: AMD VEGA 56, MESA  21.3.0
Proton Experimental

Seems to be separate problems going on. One that affects the graphics, blurring and smearing especially with camera movement, no matter the in-game settings. The other is that same issue, but when FSR is enabled there are coloured artefacts on screen. In one seen I could see the outline of the 'current objective' text that shouldn't be there (didn't get a screenshot of that)

![Screenshot from 2021-12-09 03-01-25](https://user-images.githubusercontent.com/5439355/145326980-47199370-e820-4b27-a9a2-8dc80a18c148.png)

![Screenshot from 2021-12-09 03-01-44](https://user-images.githubusercontent.com/5439355/145326987-530739ae-7795-40b8-8d99-8fa52a1f78f5.png)

**Comment by phush0** [comment]
In order to enable DLSS game needs `dxgi.nvapiHack = False` also

**Comment by jp7677** [comment]
> > Hi @phush0 could you try dxvk-nvapi from [jp7677/dxvk-nvapi#60](https://github.com/jp7677/dxvk-nvapi/pull/60) (there are binaries attached to that PR)? May be the game just stumbles over that one method when probing DLSS requirements.
> 
> Hi @jp7677 , I've attempted to do this (I believe I retrieved the correct binaries). The new nvapi doesn't cause DLSS to appear where it should in the menu, even with appropriate launch settings that function for other games. Looking at proton logs, it seems that the NvAPI workaround is still being enabled.

Could you please attach the dxvk-nvapi logs here with the nvapi binary from the PR? Please remove the old log file first because logs will be appended to the existing file.

**Comment by phush0** [comment]
> > > Hi @phush0 could you try dxvk-nvapi from [jp7677/dxvk-nvapi#60](https://github.com/jp7677/dxvk-nvapi/pull/60) (there are binaries attached to that PR)? May be the game just stumbles over that one method when probing DLSS requirements.
> > 
> > 
> > Hi @jp7677 , I've attempted to do this (I believe I retrieved the correct binaries). The new nvapi doesn't cause DLSS to appear where it should in the menu, even with appropriate launch settings that function for other games. Looking at proton logs, it seems that the NvAPI workaround is still being enabled.
> 
> Could you please attach the dxvk-nvapi logs here with the nvapi binary from the PR? Please remove the old log file first because logs will be appended to the existing file.

here working example

```
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface 0xad298d3f: Unknown function ID
дек 09 11:18:16 steam.desktop[3901]: DXVK-NVAPI experimental-6.3-20211203 (HorizonZeroDawn.exe)
дек 09 11:18:16 steam.desktop[3901]: NvAPI Output: \\.\DISPLAY1
дек 09 11:18:16 steam.desktop[3901]: NvAPI Device: Quadro RTX 5000 with Max-Q Design (470.62.16)
дек 09 11:18:16 steam.desktop[3901]: NvAPI Output: \\.\DISPLAY1
дек 09 11:18:16 steam.desktop[3901]: NvAPI Device: Intel(R) UHD Graphics (CML GT2) (21.2.5)
дек 09 11:18:16 steam.desktop[3901]: NvAPI_Initialize: OK
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface 0x33c7358c: Unknown function ID
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface 0x593e8644: Unknown function ID
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface NvAPI_D3D_ImplicitSLIControl: Not implemented method
дек 09 11:18:16 steam.desktop[3901]: NvAPI_Initialize: OK
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface NvAPI_DRS_CreateSession: Not implemented method
дек 09 11:18:16 steam.desktop[3901]: NvAPI_EnumPhysicalGPUs: OK
дек 09 11:18:16 steam.desktop[3901]: NvAPI_GPU_GetAdapterIdFromPhysicalGpu: OK
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface 0xf2400ab: Unknown function ID
дек 09 11:18:16 steam.desktop[3901]: NvAPI_EnumPhysicalGPUs: OK
дек 09 11:18:16 steam.desktop[3901]: NvAPI_GPU_GetArchInfo: OK
дек 09 11:18:16 steam.desktop[3901]: NvAPI_SYS_GetDriverAndBranchVersion: OK
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface 0xa782ea46: Unknown function ID
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface NvAPI_DRS_FindApplicationByName: Not implemented method
дек 09 11:18:16 steam.desktop[3901]: NvAPI_QueryInterface NvAPI_DRS_DestroySession: Not implemented method
дек 09 11:18:17 steam.desktop[3901]: NvAPI_Initialize: OK
дек 09 11:18:17 steam.desktop[3901]: NvAPI_D3D12_IsFatbinPTXSupported: OK
дек 09 11:18:17 steam.desktop[3901]: NvAPI_EnumPhysicalGPUs: OK
дек 09 11:18:17 steam.desktop[3901]: NvAPI_GPU_GetAdapterIdFromPhysicalGpu: OK
дек 09 11:18:17 steam.desktop[3901]: NvAPI_GPU_GetArchInfo: OK
дек 09 11:18:17 steam.desktop[3901]: NvAPI_GPU_GetAdapterIdFromPhysicalGpu: OK
дек 09 11:18:17 steam.desktop[3901]: NvAPI_D3D12_GetGraphicsCapabilities: OK
дек 09 11:18:17 steam.desktop[3901]: NvAPI_D3D12_CreateCubinComputeShaderWithName: OK
дек 09 11:19:15 steam.desktop[3901]: NvAPI_D3D12_CaptureUAVInfo: OK
дек 09 11:19:25 steam.desktop[3901]: NvAPI_D3D12_GetCudaTextureObject: OK
дек 09 11:19:25 steam.desktop[3901]: NvAPI_D3D12_LaunchCubinShader: OK
```

**Comment by jp7677** [comment]
@phush0 I'm confused by your latest posting. Does DLSS works now with the game update? The logs you posted are not from the branch like I had asked.

**Comment by phush0** [comment]
> dxgi.nvapiHack = False

It is working with stock NVAPI in Proton Experimental with `dxgi.nvapiHack = False` and `PROTON_ENABLE_NVAPI=1`

**Comment by jp7677** [comment]
Alright, then I just assume that not having DLSS was a configuration issue.

**Comment by Vaudrain** [comment]
Yes, I can confirm that not having DLSS was a configuration issue. First time I've encountered `dxgi.nvapiHack = False`, apologies.

The foliage issues confirmed by @Blitzwing1 are still present, and the addition of any upscaling (FSR or DLSS) introduces a chromatic element to the artifacting. I don't know if it's any help, but [here's a video demonstrating it](https://youtu.be/0gcdsEYfIGA), it's much easier to see in motion. The nature of the artifacts does mean that youtube compression completely crunches it though.

**Comment by kscherer** [comment]
I see similar chromatic artifacts with AMD RX580 RADV when using FidelityFX.
Even with upscaling disabled, there are still artifacts related to the foliage.

**Comment by baldpale** [comment]
I'm also affected by this bug introduced in 1.11. I'm on NVIDIA RTX 3060Ti (495), tried different Proton versions, no NVAPI enabled.

Especially the grass, but also trees are very wrongly colored and there is a lot of smearing when moving camera. I took another footage: https://streamable.com/9mvl10

I never tried to debug/dump any logs of vkd3d-proton, but let me know if I can provide anything helpful.

**Comment by noah427** [comment]
Hey I also have this issue (arch linux, AMD rx580) let me know if I can post anything to help.

**Comment by georgitsenov** [comment]
Hay, same issue here.

Ubuntu 20.04
Ryzen 5 2600
GTX 1070 Ti - Driver: 465

Tried with Proton 6.3 as well as Experimental.

**Comment by Amanoo** [comment]
Definitely got graphical artifacting, at least on foliage. It almost looks like the foliage is some transparent, with a texture layer behind it, like you might have in GIMP or Photoshop. When I move the camera, it's like looking through a window into the underlying layer, which isn't moving at all. It reminds me of the anime Gankutsuou: The Count of Monte Cristo, where a similar thing is done as a stylistic choice. It looks normal when I hit escape. Video here: https://user-images.githubusercontent.com/3501477/145696479-6e358c49-30d2-4444-91fb-0b229339cf23.mp4

Another bug, when I try to use the new FSR upscaling option from within the game, all the 3D graphics turn black. I can see the HUD and the menus, but I can't see the world.

Game version 1.11
EndeavourOS
Kernel 5.15.7-arch-1
7700K
GTX1080ti
Nvidia 495.44
Proton 6.21 GE 2

**Comment by kscherer** [comment]
On ProtonGE 6.21-GE2 AMD RX580 adding RADV_DEBUG=llvm to steam command fixed the graphical artifacts with foliage. This workaround hasn't affected performance and the game is running well.

**Comment by Amanoo** [comment]
> On ProtonGE 6.21-GE2 AMD RX580 adding RADV_DEBUG=llvm to steam command fixed the graphical artifacts with foliage. This workaround hasn't affected performance and the game is running well.

Do you, or anyone else, know if there's an Nvidia equivalent to this?

**Comment by hakzsam** [comment]
@kscherer What issue is that? Could you give more information, hardware, mesa version, screenshot etc please?

**Comment by kscherer** [comment]
The screenshots from @Blitzwing1 above show the graphical artifacts related to foliage. With FSR there are extra colors. Without FSR there is also graphical corruption. It is this specific issue that is "fixed" by adding RADV_DEBUG=llvm to the Steam command. With that addition the game runs great and I haven't noticed any other issues.

My system is RX580 running Proton 6.21-GE2 on Ubuntu 21.10 with mesa 21.2.2

I learned about this workaround here: https://www.protondb.com/app/1151640

**Comment by sebastiencaty** [comment]
I know it's not much help but on a 5700XT, Proton Experimental and current mesa main branch it works fine. With or without FSR and both version 1.11/1.11.1

I can try running on another mesa release. Might possibly be already fixed.

**Comment by hakzsam** [comment]
No, it's very useful! If you could try to reproduce with the same version (Mesa 21.2.2) that would be awesome. It could be a Polaris specific issue or something we fixed since Mesa 21.2.2.

**Comment by sebastiencaty** [comment]
Tried mesa 21.2.2 with both Proton 6.3-8 and Experimental. Can't reproduce on NAVI10.

**Comment by dgalli1** [comment]
I'm on Mesa 21.2.5 with Proton Experimental and i experience the same foliage issue.
Using llvm works fine for me aswell.

GPU is a AMD VEGA 56

**Comment by kakra** [comment]
> It could be a Polaris specific issue or something we fixed since Mesa 21.2.2.

This is probably not an AMD-specific issue, it happens in exactly the same way on NVIDIA. I think the debug mode just changes some context of how vkd3d, the game, and the drivers interact so it may be some ordering or locking issue. I'm not sure, tho, if this is related to a recent game update (I think they FSR lately), or a vkd3d/Proton update (didn't run the game in a while) - the title indicates it's due to a game update. Since both NVIDIA and AMD show the issue, it's unlikely that a Mesa update would fix it - if it does, it's probably rather a "hide" than a "fix" (similar to how the RADV debug mode hides the bug). So it's probably more likely that a vkd3d improvement properly fixed this.

If using FSR on NVIDIA, the foliage is covered in LSD-like rainbow colors drawn over a black background but it still has the same "smearing" effects we can see without FSR where it seems that the game just draws alpha textures over non-cleared buffers, leaving behind traces of the previous frame which then slowly fades away. While only foliage seems to be affected (which probably has a lot of alpha texels), it seems that lighting and illumination is sometimes also way off - but that may just be my personal experience over the black foliage with the funny colorful LSD effects... :-D

**Comment by Blitzwing1** [comment]
> No, it's very useful! If you could try to reproduce with the same version (Mesa 21.2.2) that would be awesome. It could be a Polaris specific issue or something we fixed since Mesa 21.2.2.

I can say that it likely isn't something that was fixed in versions beyond 21.2.2 as I tested it on 21.3.0 and had the exact same issues, nor do I think is it related to Polaris as I'm on vega and it still happened, plus others seem to have the problem on Nvidia chips. 

Unfortunatley I had x11 bug to all heck on my desktop to the point where i had to reinstall the whole os, I'm currently transferring HZD from my laptop back over where I can test 21.3.1, with and without the "RADV_DEBUG=llvm" ENV variable pointed out by @kscherer and report back.

**Comment by Amanoo** [comment]
> I'm not sure, tho, if this is related to a recent game update (I think they FSR lately), or a vkd3d/Proton update (didn't run the game in a while) - the title indicates it's due to a game update.

Pretty sure it's a game update, since I was actively playing this game on a Glorious Eggroll version of Proton, didn't change anything other than update the game, and suddenly had this bug. And I have a 1080ti, so no AMD Mesa whatever for me. May have been something already broken in vkd3d and that simply got exposed by the update for all I know, though. Or vkd3d exposing a bug in the game. You tell me.

**Comment by dgalli1** [comment]
Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.

**Comment by Amanoo** [comment]
> Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.

How did you downgrade? I've been trying DepotDownloader and I just couldn't get it to work. It kept downloading 1.11 files anyway.

**Comment by Blitzwing1** [comment]
So, I've tried both 21.2.2 and 21.3.1 both still have the same issues as described in other comments, smearing and blurring without up-scaling, and when up-scaling is enabled coloured artefacts too. With "RADV_DEBUG=llvm" set then it cures those issues and seemingly allows up-scaling with FSR to work as well. 

Seems like whatever issue is happening it doesn't affect AMD's official shader compiler, but obviously impacts RADV and Nvidia's drivers in the same way.

**Comment by SBKarr** [comment]
> > Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.
> 
> How did you downgrade? I've been trying DepotDownloader and I just couldn't get it to work. It kept downloading 1.11 files anyway.

Depots/manifests to downgrade

`dotnet DepotDownloader.dll -app 1151640 -depot 1151642 -manifest 2110572734960666938 -username <name> -password <password>`
`dotnet DepotDownloader.dll -app 1151640 -depot 1151641 -manifest 8564283306590138028 -username <name> -password <password>`

Links to SteamDB:
https://steamdb.info/depot/1151642/history/?changeid=M:6986189025896008261
https://steamdb.info/depot/1151641/history/?changeid=M:3712644444013817231

Updated files:
HorizonZeroDawn.exe
LocalCacheDX12/HashDB.bin
LocalCacheDX12/ShaderLocationDB.bin
Packed_DX12/Patch.bin

No artefacts after downgrade
![2021-12-20 07-31-31](https://user-images.githubusercontent.com/2579526/146694950-439bed1c-b8bc-4cc5-9036-e02d6471376e.png)
e

**Comment by Amanoo** [comment]
> > > Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.
> > 
> > 
> > How did you downgrade? I've been trying DepotDownloader and I just couldn't get it to work. It kept downloading 1.11 files anyway.
> 
> Depots/manifests to downgrade
> 
> `dotnet DepotDownloader.dll -app 1151640 -depot 1151642 -manifest 2110572734960666938 -username <name> -password <password>` `dotnet DepotDownloader.dll -app 1151640 -depot 1151641 -manifest 8564283306590138028 -username <name> -password <password>`
> 
> Links to SteamDB: https://steamdb.info/depot/1151642/history/?changeid=M:6986189025896008261 https://steamdb.info/depot/1151641/history/?changeid=M:3712644444013817231
> 
> Updated files: HorizonZeroDawn.exe LocalCacheDX12/HashDB.bin LocalCacheDX12/ShaderLocationDB.bin Packed_DX12/Patch.bin
> 
> No artefacts after downgrade ![2021-12-20 07-31-31](https://user-images.githubusercontent.com/2579526/146694950-439bed1c-b8bc-4cc5-9036-e02d6471376e.png)

I was already doing that. Seems to me now that while I did, I wasn't actually launching from my downgraded game folder, but from my updated game folder. I'm now on 1.10 again. Nice. Now there's still one problem, how do you prevent Steam from updating the game again? Because apparently that was what I was (and still am) actually struggling with.

**Comment by georgitsenov** [comment]
> dotnet DepotDownloader.dll

Awesome!

Can you provide a more detailed instructions on how to do this magic?

**Comment by georgitsenov** [comment]
> > > Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.
> > 
> > 
> > How did you downgrade? I've been trying DepotDownloader and I just couldn't get it to work. It kept downloading 1.11 files anyway.
> 
> Depots/manifests to downgrade
> 
> `dotnet DepotDownloader.dll -app 1151640 -depot 1151642 -manifest 2110572734960666938 -username <name> -password <password>` `dotnet DepotDownloader.dll -app 1151640 -depot 1151641 -manifest 8564283306590138028 -username <name> -password <password>`
> 
> Links to SteamDB: https://steamdb.info/depot/1151642/history/?changeid=M:6986189025896008261 https://steamdb.info/depot/1151641/history/?changeid=M:3712644444013817231
> 
> Updated files: HorizonZeroDawn.exe LocalCacheDX12/HashDB.bin LocalCacheDX12/ShaderLocationDB.bin Packed_DX12/Patch.bin
> 
> No artefacts after downgrade ![2021-12-20 07-31-31](https://user-images.githubusercontent.com/2579526/146694950-439bed1c-b8bc-4cc5-9036-e02d6471376e.png)

Awesome work!
Can we get a more step-by-step details on how to do this magic?

**Comment by Amanoo** [comment]
> Awesome work! Can we get a more step-by-step details on how to do this magic?

Well it's really just down to downloading DepotDownloader (https://github.com/SteamRE/DepotDownloader), and then running the aforementioned 2 commands in your DepotDownloader directory. 2 folders will be created with some files inside, among which those 4 files. One will contain the exe, the other folder the bin files. Copy them over to your game folder and you've now downgraded your game.

I recommend backing stuff up, Steam can be a bit annoying, trying to force game updates down your throat. If you have a game folder somewhere that you know works, you can always go back.

**Comment by georgitsenov** [comment]
> > Awesome work! Can we get a more step-by-step details on how to do this magic?
> 
> Well it's really just down to downloading DepotDownloader (https://github.com/SteamRE/DepotDownloader), and then running the aforementioned 2 commands in your DepotDownloader directory. 2 folders will be created with some files inside, among which those 4 files. One will contain the exe, the other folder the bin files. Copy them over to your game folder and you've now downgraded your game.
> 
> I recommend backing stuff up, Steam can be a bit annoying, trying to force game updates down your throat. If you have a game folder somewhere that you know works, you can always go back.

I suspect that Steam will override the files when I try to start the game and they will be updated (broken again) - am I correct?

**Comment by Amanoo** [comment]
> I suspect that Steam will override the files when I try to start the game and they will be updated (broken again) - am I correct?

It did update it at first, yes. But I just replaced the files again, and Steam thought it was still the updated version when it wasn't anymore. No idea when Steam will notice that my version is out of date, though. It very much might try to update again later.

**Comment by georgitsenov** [comment]
> > I suspect that Steam will override the files when I try to start the game and they will be updated (broken again) - am I correct?
> 
> It did update it at first, yes. But I just replaced the files again, and Steam thought it was still the updated version when it wasn't anymore. No idea when Steam will notice that my version is out of date, though. It very much might try to update again later.

I can confirm the fix and would like to add a few points.
The game ran without issues and it did not re-update the "reverted" files.

In order to avoid downloading dozens of gigabytes you can create two files:
file1:
`HorizonZeroDawn.exe`
file2:
```
LocalCacheDX12/HashDB.bin
LocalCacheDX12/ShaderLocationDB.bin
Packed_DX12/Patch.bin
```
and after that run the commands:
```
$ dotnet DepotDownloader.dll -app 1151640 -depot 1151642 -manifest 2110572734960666938 -username ${USERNAME} -filelist file1

$ dotnet DepotDownloader.dll -app 1151640 -depot 1151641 -manifest 8564283306590138028 -username ${USERNAME} -filelist file2
```
The ${USERNAME} variable should be set with your Steam username. You will be prompted for password as well as the 2FA code.

**Comment by beltet** [comment]
> Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.

How do you downgrade? I haven't found a good answer

**Comment by mkfelidae1** [comment]
> > Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.
> 
> How do you downgrade? I haven't found a good answer

first you will need to install dotnet runtime on your machine (assuming you do not already have it).  You will need dotnet runtime 5.0 or newer specifically.  if you search dotnet runtime linux there should be a link from microsoft.com in the first page of your results walking you through installation for your specific linux distro (or its base distro, in my case ubuntu 20.04).

The next thing you will need is the DepotDownloader referenced above in @Amanoo 's post.

after that you should follow the download steps in @georgitsenov 's last post.  I recommend making the file lists he refers to in nano, it was easy enough as I was already in a bash shell for most of the dotnet installation process.

Finally you need to replace the existing HZD files in your steam folder with the ones you have just downloaded.  launch the game and you are golden.  Hope this helps.

**Comment by beltet** [comment]
> 

Thanks for the reply!
Will try it out later today!

**Comment by beltet** [comment]
> > > Jup i downgraded the game back to 1.10 and it worked flawless again, its definitely the update.
> > 
> > 
> > How do you downgrade? I haven't found a good answer
> 
> first you will need to install dotnet runtime on your machine (assuming you do not already have it). You will need dotnet runtime 5.0 or newer specifically. if you search dotnet runtime linux there should be a link from microsoft.com in the first page of your results walking you through installation for your specific linux distro (or its base distro, in my case ubuntu 20.04).
> 
> The next thing you will need is the DepotDownloader referenced above in @Amanoo 's post.
> 
> after that you should follow the download steps in @georgitsenov 's last post. I recommend making the file lists he refers to in nano, it was easy enough as I was already in a bash shell for most of the dotnet installation process.
> 
> Finally you need to replace the existing HZD files in your steam folder with the ones you have just downloaded. launch the game and you are golden. Hope this helps.

Thanks for the reply!
I will try it out later today

**Comment by PercentBoat4164** [comment]
I wrote a bash shell script that may help others who need to perform this downgrade in the future. I am unfamiliar with the rules or etiquette with sharing files in comment strings on Github. Should I just paste it here, or push it to a repository on my Github and comment a link to it here, or should I take some other third course of action?

**Comment by PercentBoat4164** [comment]
I have uploaded a script that will automatically downgrade Horizon Zero Dawn for those who are having trouble in the future. It can be found [here](https://github.com/PercentBoat4164/HZDDowngrader).

**Comment by abhishekpapanna** [comment]
The game is crashing after replacing with these files. The message just says Unfortunately the game has crashed, do you want to send a crash report.

**Comment by Amanoo** [comment]
> The game is crashing after replacing with these files. The message just says Unfortunately the game has crashed, do you want to send a crash report.

Try launching it a few more times. My game also crashes a lot at launch these days. Once you've been in the game story for a few minutes, it's stable.

**Comment by ipkpjersi** [comment]
Hi @HansKristian-Work does this fix in #962 also fix the grass texture bug (grass smearing etc, the bug seen here: https://github.com/HansKristian-Work/vkd3d-proton/issues/941#issuecomment-990314379)?

Thanks.

**Comment by Blitzwing1** [comment]
> Hi @HansKristian-Work does this fix in #962 also fix the grass texture bug (grass smearing etc, the bug seen here: [#941 (comment)](https://github.com/HansKristian-Work/vkd3d-proton/issues/941#issuecomment-990314379))?
> 
> Thanks.

Hi, I know you didn’t direct the question at me but if I can save busy maintainers their time I will try. I’ve only had time to boot up the game run around a bit, try it out on my AMD desktop and Nvidia laptop, with and without FSR on the AMD machine and with and without DLSS on the Nvidia and I’m fairly sure it is working just like before the 1.11 update to the game.

tl;dr seems all good now.

**Comment by ipkpjersi** [comment]
That sounds great to me, hopefully I can test it out when it comes to Proton Experimental and/or ProtonGE and confirm that the grass bug is fixed for me too.

**Comment by ipkpjersi** [comment]
I can confirm that the latest vkd3d-proton fix in #962 has fixed not only the grass texture smearing but also the FSR black screen bug I was experiencing. I'm getting better graphics and better performance now thanks to the fix.

**Comment by Amanoo** [comment]
Are there some instructions somewhere on adding the fixed version to Proton?

**Comment by kakra** [comment]
Proton Experimental usually carries the latest dxvk and vkd3d branches when it is updated. I just tried and can confirm: Graphics rendering is working properly again, and the game also loads proper textures again instead of super-lowres textures with reduced or even missing geometry - all good again.

**Comment by Amanoo** [comment]
I just updated HZD and downloaded Proton GE 7.0rc6 which has just been released. I can no longer look into another dimension by looking at grass, whether with FSR on or with FSR off. Issue resolved. 

It still looks a little bit trash with FSR on, but that's probably just FSR itself making it look a bit grainy through some sort of oversharpening. Simple upscaling makes it a bit blurry instead of grainy, since it actually does too little while FSR overcompensates. I think I prefer Proton's built in FSR implementation.
